servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2Setting.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+/**
+ * Represents an <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>
+ * &lt;Identifier, Value&gt; pair.
+ */
+public interface H2Setting {
```

Comment:
this PR can be decoupled in to multiple PRs, but first wanted to get feedback on the general approach.